src/config/mod.rs

```diff
@@ -3,29 +3,36 @@
 //! Reads a TOML config file for the IPC Agent and deserializes it in a type-safe way into a
 //! [`Config`] struct.
 
+mod deserialize;
+
 use std::collections::HashMap;
-use std::fmt::Formatter;
 use std::fs;
-use std::str::FromStr;
+use std::net::SocketAddr;
 
 use anyhow::Result;
 use fvm_shared::address::Address;
 use ipc_sdk::subnet_id::SubnetID;
-use serde::de::{Error, SeqAccess};
-use serde::{Deserialize, Deserializer};
+use serde::Deserialize;
 use url::Url;
+use crate::config::deserialize::{deserialize_subnet_id, deserialize_accounts};
 
 /// The top-level struct representing the config. Calls to [`Config::from_file`] deserialize into
 /// this struct.
 #[derive(Deserialize)]
 pub(crate) struct Config {
+    pub server: Server,
     pub subnets: HashMap<String, Subnet>,
 }
 
+#[derive(Deserialize, Clone)]
+pub struct Server {
+    pub json_rpc_address: SocketAddr,
+}
+
 /// Represents a subnet declaration in the config.
 #[derive(Deserialize)]
 pub struct Subnet {
```

Comment:
What about separating these into a `subnet.rs` and `server.rs` and move all relevant constants for them to these files? Like for instance `DEFAULT_JSON_RPC_SERVER_ENDPOINT`

Reply:
I did that initially, but then I realized they are actually quite small. I was thinking of breaking into different files when they grow larger.

Reply:
+1 to separating these into their own modules now.